toolchain/lower/function_context.cpp

```diff
@@ -116,4 +126,17 @@ auto FunctionContext::CopyValue(SemIR::TypeId type_id, SemIR::InstId source_id,
   }
 }
 
+auto FunctionContext::Inserter::InsertHelper(
+    llvm::Instruction* inst, const llvm::Twine& name, llvm::BasicBlock* block,
+    llvm::BasicBlock::iterator insert_pt) const -> void {
+  llvm::StringRef base_name = (inst_namer_ && !inst->getType()->isVoidTy())
+                                  ? inst_namer_->GetUnscopedNameFor(inst_id_)
+                                  : "";
+  IRBuilderDefaultInserter::InsertHelper(
+      inst,
+      (!base_name.empty() && !name.isTriviallyEmpty()) ? base_name + "." + name
+                                                       : base_name + name,
+      block, insert_pt);
```

Comment:
I know `Twine` fights hard, but maybe something like this would work?

```suggestion
  auto prefix_name = [&] -> llvm::Twine {
    if (inst_namer_ && !inst->getType()->isVoidTy() &&
        !name.isTriviallyEmpty()) {
      return llvm::Twine(inst_namer_->GetUnscopedNameFor(inst_id_)) + "." + name;
    }
    
    return name;
  };
  IRBuilderDefaultInserter::InsertHelper(
      inst, prefix_name(), block, insert_pt);
```

Reply:
I don't think that works, sadly -- the twine returned by the lambda has another twine as a child, and that inner twine is destroyed too early.

Switched to a different approach.